Client/Frontend/Browser/Tabs/Views/TabDisplayView.swift

```diff
@@ -203,27 +209,67 @@ class TabDisplayView: UIView,
             guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: TabCell.cellIdentifier, for: indexPath) as? TabCell
             else { return UICollectionViewCell() }
 
-            let tabState = TabCellState(isSelected: false,
-                                        isPrivate: false,
-                                        isFxHomeTab: false,
-                                        tabTitle: "\(indexPath.row)",
-                                        url: nil,
-                                        screenshot: nil,
-                                        hasHomeScreenshot: false,
-                                        margin: 0)
-            cell.configure(with: tabState, theme: theme)
+            let tabState = tabTrayState.tabs[indexPath.row]
+            cell.configure(with: tabState, theme: theme, delegate: self)
             return cell
         }
     }
 
     @objc
     func toggleInactiveTab() {
-        toggleInactiveTabSection(hasExpanded: !state.isInactiveTabsExpanded)
+        toggleInactiveTabSection(hasExpanded: !tabTrayState.isInactiveTabsExpanded)
         collectionView.collectionViewLayout.invalidateLayout()
     }
 
     private func toggleInactiveTabSection(hasExpanded: Bool) {
-        state.isInactiveTabsExpanded = hasExpanded
+        tabTrayState.isInactiveTabsExpanded = hasExpanded
         collectionView.reloadData()
     }
+
+    // MARK: - TabCellDelegate
+    func tabCellDidClose(_ cell: TabCell) {
+        if let indexPath = collectionView.indexPath(for: cell) {
+            store.dispatch(TabTrayAction.closeTab(indexPath.row))
+        }
```

Comment:
Nit: might be slightly cleaner to use `guard` here?

```suggestion
        guard let indexPath = collectionView.indexPath(for: cell) else { return }
        store.dispatch(TabTrayAction.closeTab(indexPath.row))
```